services/121-service/src/notifications/message.service.ts

```diff
@@ -310,31 +311,85 @@ export class MessageService {
     }
   }
 
-  private async getNotificationText(
-    language: string,
+  private async getTemplateMessageTextOrFallback({
+    programId,
+    messageTemplateKey,
+    preferredLanguage,
+  }: {
+    programId: number;
+    messageTemplateKey: string;
+    preferredLanguage: string;
+  }): Promise<string> {
+    return (
+      (
+        await this.getMessageTemplateForLanguageOrFallback(
+          programId,
+          messageTemplateKey,
+          preferredLanguage,
+        )
+      ).message ?? ''
+    );
+  }
+
+  private async getTemplateContentSidOrFallback({
+    programId,
+    messageTemplateKey,
+    preferredLanguage,
+  }: {
+    programId: number;
+    messageTemplateKey: string;
+    preferredLanguage: string;
+  }): Promise<string> {
+    const messageTemplate = await this.getMessageTemplateForLanguageOrFallback(
+      programId,
+      messageTemplateKey,
+      preferredLanguage,
+    );
+    if (!messageTemplate.contentSid) {
+      throw new Error(
+        `No contentSid found for message template key: ${messageTemplateKey}, programId: ${programId}, language: ${preferredLanguage}`,
+      );
+    }
+    return messageTemplate.contentSid;
+  }
+
+  private async getMessageTemplateForLanguageOrFallback(
+    programId: number,
     messageTemplateKey: string,
-    programId?: number,
-  ): Promise<string> {
+    preferredLanguage: string,
+  ): Promise<{ message: string | null; contentSid: string | null }> {
     const messageTemplates = await this.messageTemplateRepo.findBy({
       program: { id: programId },
       type: messageTemplateKey,
     });
 
+    // Try to find template in preferred language
     const notification = messageTemplates.find(
-      (template) => template.language === language,
+      (template) => template.language === preferredLanguage,
     );
     if (notification) {
-      return notification.message;
+      return {
+        message: notification.message,
+        contentSid: notification.contentSid,
+      };
     }
 
+    // Fallback to default language
     const fallbackNotification = messageTemplates.find(
       (template) => template.language === this.fallbackLanguage,
     );
     if (fallbackNotification) {
-      return fallbackNotification.message;
+      return {
+        message: fallbackNotification.message,
+        contentSid: fallbackNotification.contentSid,
+      };
     }
 
-    return '';
+    // No template found
```

Comment:
```suggestion
    // No template found
    // This can happen on projects that have Whatsapp and templates configured incorrectly.
    // TODO: Add feature to pro-actively inform admins of this.
```